src/services/link/hooks/index.js

```diff
@@ -1,14 +1,37 @@
 const hooks = require('feathers-hooks-common');
 const { authenticate } = require('@feathersjs/authentication');
 const globalHooks = require('../../../hooks');
+const logger = require('../../../logger');
+
+const preventDuplicates = async (hook) => {
+	const linkData = hook.data;
+	// get all links to the same target
+	const currentLinks = await hook.app.service('link').find({ query: { target: linkData.target } });
+
+	// check if link to specified target already exists or new link is forced
+	if (currentLinks && currentLinks.total > 0 && !linkData.forceNew) {
```

Comment:
- missed to check linkData exist before requesting it's properties
- check currentLinks.data directly (if currentLinks) instead of total. `Array.isArray(...)`, then for `length !== 0`